src/BlockEntities/ChestEntity.cpp

```diff
@@ -217,11 +219,33 @@ void cChestEntity::DestroyWindow()
 
 
 
+class cFindSittingCat :
+	public cEntityCallback
+{
+	virtual bool Item(cEntity * a_Entity) override
+	{
+		return (
+			(a_Entity->GetEntityType() == cEntity::etMonster) &&
+			(static_cast<cMonster*>(a_Entity)->GetMobType() == eMonsterType::mtOcelot) &&
+			(static_cast<cOcelot*>(a_Entity)->IsSitting())
+		);
+	}
+};
+
+
+
+
+
 bool cChestEntity::IsBlocked()
 {
-	// TODO: cats are an obstruction
+	cFindSittingCat FindSittingCat;
 	return (
 		(GetPosY() >= cChunkDef::Height - 1) ||
-		!cBlockInfo::IsTransparent(GetWorld()->GetBlock(GetPosX(), GetPosY() + 1, GetPosZ()))
+		!cBlockInfo::IsTransparent(GetWorld()->GetBlock(GetPosX(), GetPosY() + 1, GetPosZ())) ||
+		((GetWorld()->GetBlock(GetPosX(), GetPosY() + 1, GetPosZ()) == E_BLOCK_AIR) && !GetWorld()->ForEachEntityInBox(cBoundingBox(Vector3d(GetPosX(), GetPosY() + 1, GetPosZ()), 1, 1), FindSittingCat))
```

Comment:
This line is rather long, split it in two, and use an additional level of indent:
```cpp
return (
  cond1 ||
  cond2 ||
  (
    cond3 &&
    cond4
  )
);
```